ledger-core/conveyor/smachine/slot_machine_test.go

```diff
@@ -90,3 +90,86 @@ func TestSlotMachine_AddSMAndMigrate(t *testing.T) {
 	assert.True(t, s.ok)
 	assert.False(t, s.notOk)
 }
+
+type smTestRestoreStep struct {
+	smachine.StateMachineDeclTemplate
+
+	wasContinued bool
+	wasMigrated  bool
+}
+
+func (s *smTestRestoreStep) GetInitStateFor(_ smachine.StateMachine) smachine.InitFunc {
+	return s.stepInit
+}
+
+func (s *smTestRestoreStep) GetStateMachineDeclaration() smachine.StateMachineDeclaration {
+	return s
+}
+
+func (s *smTestRestoreStep) stepInit(ctx smachine.InitializationContext) smachine.StateUpdate {
+	ctx.SetDefaultMigration(s.migrate)
+	return ctx.Jump(s.stepWaitInfinity)
+}
+
+func (s *smTestRestoreStep) stepWaitInfinity(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	return ctx.Sleep().ThenRepeat()
+}
+
+func (s *smTestRestoreStep) migrate(ctx smachine.MigrationContext) smachine.StateUpdate {
+	s.wasMigrated = true
+	affectedStep := ctx.AffectedStep()
+	affectedStep.Transition = func(ctx smachine.ExecutionContext) smachine.StateUpdate {
```

Comment:
remove it and replace with 
return ctx.RestoreStep(ctx.AffectedStep())

Reply:
done